src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1521,6 +1521,28 @@ private bool CheckIsVariable(SyntaxNode node, BoundExpression expr, BindValueKin
                 return true;
             }
 
+            var conditional = expr as BoundConditionalOperator;
+            if (conditional != null)
```

Comment:
Consider using `switch (expr.Kind) { ... }` for this `if` and the two `if`s above.